interface/orders/receive_hl7_results.inc.php

```diff
@@ -430,17 +478,133 @@ function match_patient($ptarr)
             array($in_ss, $in_fname, $in_lname, $in_dob)
         );
         if (!empty($tmp['pid'])) {
-              $patient_id = -1;
+            $patient_id = -1;
         }
     }
 
     return $patient_id;
 }
 
+/**
+ * Look for a lab matching the given XCN field from some segment.
+ *
+ * @param array $seg MSH seg identifying a provider.
+ * @return mixed        TRUE, or FALSE if no match.
+ */
+function match_lab(&$hl7, $send_acct, $lab_acct = '', $lab_app = '', $lab_npi = '')
+{
+    if (empty($hl7)) {
+        return false;
+    }
+    $d0 = "\r";
+    $d1 = substr($hl7, 3, 1); // typically |
+
+    $segs = explode($d0, $hl7);
+    $a = explode($d1, $segs[0]);
+    if ($a[0] != 'MSH') {
+        unset($segs);
+        return false;
+    }
+
+    unset($segs);
+    if ($lab_npi == '1891752424' || strtoupper($lab_npi) == 'AMMON') {
+        if (strtoupper(trim($a[5])) == strtoupper(trim($send_acct))) {
+            $srch = '|' . strtoupper(trim($send_acct)) . '-';
+            $hl7 = str_replace($srch, "|", $hl7);
+            return true;
+        }
+        return false;
+    }
+
+    if (
+        strtoupper(trim($a[5])) == strtoupper(trim($send_acct)) ||
+        strtoupper(trim($a[3])) == strtoupper(trim($lab_acct)) ||
+        strtoupper(trim($a[2])) == strtoupper(trim($lab_app))
+    ) {
+        return true;
+    }
+
+    return false;
+}
+
+// lookup code from configuration to validate codes after order submit ie manual/batch
+function lookupTestCode($labid, $procedure_code)
+{
+
+    $query = "SELECT procedure_type_id, procedure_code, procedure_type, name, transport " .
+        "FROM procedure_type WHERE " .
+        "lab_id = ? AND " .
+        "(procedure_type LIKE 'ord' OR procedure_type LIKE 'pro') AND " .
+        "activity = 1 AND procedure_code = ? " .
+        "LIMIT 1";
+    $res = sqlQuery($query, array($labid, $procedure_code));
+
+    return $res;
+}
+
+// create encounter
+function create_encounter($pid, $provider_id = '', $order_date, $lab_name)
+{
+    global $orphanLog;
+    $conn = $GLOBALS['adodb']['db'];
+    $encounter = $conn->GenID("sequences");
+    addForm(
+        $encounter,
+        "Auto Generated Lab Encounter",
+        sqlInsert("INSERT INTO form_encounter SET " .
+            "date = ?, " .
+            "onset_date = '', " .
+            "reason = 'Generated encounter for " . strtoupper($lab_name) . " result', " .
```

Comment:
```
"reason = 'Generated encounter for " . add_escape_custom(strtoupper($lab_name)) . " result', " .
```

Reply:
actually, would not do this since a `?` in this variable would break this (ie. can not mix binding and add_escape_custom()). Instead should make this entire string a bind (`?`) and then in the binded array below, place
```
array(date('Y-m-d H:i:s', strtotime($order_date)), "Generated encounter for " . strtoupper($lab_name) . " result", $pid, $encounter, $provider_id)
```